grunttasks/l10n-translate-verify.js

```diff
@@ -0,0 +1,94 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  grunt.registerTask('l10n-translate-verify', 'Run the XSS parser on the generated l10n files', [
+    'clean',
+    'selectconfig:dist',
+    'l10n-generate-pages',
+    'l10n-json-to-html',
+    'generate-valid-urls',
+    'verify-urls'
+    ]
+  );
+
+  grunt.registerMultiTask('verify-urls', 'Verify whether the translated urls are valid', function(){
+    var fileArray = this.files[0].src;
+    var contents = grunt.file.read(fileArray.shift());
+    var urlArray = JSON.parse(contents);
+    var hasErrors = 0;
+    var urlRegex = /(href="[\w/:\\\.%()]+")/ig;
+    // Extract the href from each file and check if it belongs to
+    // a valid representation of the url. Flag it if not.
+    fileArray.forEach(function (src) {
+      var html = grunt.file.read(src);
+      var urlList = html.match(urlRegex);
+      // Check if there were any urls found, .match() might return null
+      if (urlList !== null) {
+        urlList.forEach(function(url){
+          if (urlArray.indexOf(url) === -1) {
+            grunt.log.error('Found an url which should not be there', url, src);
+            hasErrors = hasErrors + 1;
+          }
+        });
+      }
+    });
+    if (hasErrors !== 0) {
+      grunt.fail.warn('Found ' + hasErrors + ' ' + grunt.util.pluralize(hasErrors, 'error/errors') + ' in ' + fileArray.length + ' files');
+    } else {
+      grunt.log.writeln('Checked ' + fileArray.length + ' files for valid hrefs and found nothbing wrong :D');
```

Comment:
its nothbing, not-having-bing, not-h-bing. y u do this to me, Peter? :japanese_goblin: